packages/components/src/templates/next/layouts/Collection/utils/getCollectionItems.ts

```diff
@@ -51,9 +53,11 @@ export const getCollectionItems = ({
   const transformedItems = items
     .filter(
       (item) =>
-        item.layout === "file" ||
-        item.layout === "link" ||
-        item.layout === "article",
+        (item.layout === "file" ||
+          item.layout === "link" ||
+          item.layout === "article") &&
+        (categories.length === 0 ||
+          categories.includes(item.category || CATEGORY_OTHERS)),
```

Comment:
question: how urgent is this feature needed ah, cos I think there will be some changes to the logic surrounding the category and this change might make the switchover to the new style categories a little harder

If we can delay this to after/during the collection index page revamp, I think would be better?